include/soci/soci-backend.h

```diff
@@ -366,7 +366,7 @@ class session_backend
     {
         return "select table_name as \"TABLE_NAME\""
             " from information_schema.tables"
-            " where table_schema = 'public'";
+            " where table_schema = '" + schema_name_ + "'";
```

Comment:
Shouldn't we use a parameter for this? I'm not sure if the schema name will always be under program control and if it isn't, then this is a flagrant SQL injection problem.

Reply:
The variable _should_ be managed by backends - but I totally agree that it should be a parameter, I was just a bit in doubt how to achieve this while not changing the API and contracts.